src/vite/utils.ts

```diff
@@ -40,10 +39,28 @@ export const handleDevToolsViteRequest = (
     const dataToParse = Buffer.concat(chunks);
    try { const parsedData = JSON.parse(dataToParse.toString());
     cb(parsedData);} 
-    // eslint-disable-next-line no-empty
+      // eslint-disable-next-line no-empty
     catch(e){
 
     }
     res.write("OK");
   });
 };
+
+export function checkPath(routePath: string, extensions = [".tsx", ".jsx", ".ts", ".js"]) {
+  // Check if the path exists as a directory
+  if (fs.existsSync(routePath) && fs.lstatSync(routePath).isDirectory()) {
+    return { validPath: routePath, type: "directory" } as const;
+  }
+
+  // Check if the path exists as a file with one of the given extensions
+  for (const ext of extensions) {
+    const filePath = `${routePath}${ext}`;
+    if (fs.existsSync(filePath) && fs.lstatSync(filePath).isFile()) {
```

Comment:
I think the 2nd check is not needed because a folder wouldn't have an extension, I mean you could write it in but in practice I don't think anyone is doing something like `folder.jsx/file.jsx` or at least I'd hope not, or maybe even move this check into the return below, so `type: fs.lstatSync(filePath).isFile() ? "file" : "directory"`